### PR TITLE
Add support for configuring the nunjucks template path of descriptions

### PR DESCRIPTION
**Pull request by joshyork**
TLDR; Adds support for accessing templates outside of the directory being used for the styleguide.

When writing descriptions it would be nice to be able to access templates that live outside of the styleguide directory. This PR adds support for specifying a path to use when compiling the descriptions.

Imagine the following file structure:

```
| -- src
|  |-- assets
|  |  |-- somewhere under here a btn.md to document _btn.scss
|  |-- includes
|  |  |-- macros.njk
|  |-- styleguide
|  |  |-- all the styleguide stuff templates and assets.
```

Currently it's not possible to import `macros.njk` from `btn.md` because they are in adjacent folders and you can't use relative paths with `nunjucks.renderString`. This is assuming that the styleguide directory is being used as the options.path.src and you are not wanting to use emo-gen to compile the static pages of the site.

For the structure above this PR would allow you to specify the `src` path and then import the macros like 

```
{% import 'includes/macros.njk' as macros %}
```

This PR is currently just at a rough proof of concept stage. There are other methods of providing the same functionality that may be better. If there is interest in this PR I can clean it up and add documentation.

### CODE REVIEW

**Comment by martinjunior** [comment]
@JoshYork I like the idea. I'm going to do a round of testing, then I'll incorporate it.
